### PR TITLE
Add editorjs renderer

### PR DESCRIPTION
**Pull request by krzysztofwolski**
Support for new type of renderer (editorjs)

### Screenshots

image

image

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [x] Changes are mentioned in the changelog.

### Test Environment Config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1587 Render EditorJS format in storefront](https://app.clickup.com/t/2549495/SALEOR-1587)

File: src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          tools,
+        });
+      }
+
+      return editor.current?.destroy;
+    },
+    // Rerender editor only if changed from undefined to defined state
+    [data === undefined]
+  );
+  React.useEffect(() => editor.current?.destroy, []);
```
Comment (jwm0) [inline src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx:62]: Duplicate (destructor is already defined above)

**Review by orzechdev — COMMENTED** [review]
Cypress tests fail

File: src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          onReady,
+          readOnly: true,
+          tools,
+        });
+      }
+
+      return editor.current?.destroy;
+    },
+    // Rerender editor only if changed from undefined to defined state
+    [data === undefined]
```
[the remark this answers is not in the document]
Comment (dominik-zeglen) [reply src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx:60]: Can be undefined if loading. It serves it's purpose in dashboard so we can display outline etc, but we don't have anything like this here

File: src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx:62]: It's connected to issue above, probably safe to remove it

Comment (jwm0) [reply src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx:62]: It's not necessary no matter the case - all `useEffect` hooks call return fn on unmount

File: src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jwm0) [reply src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx:60]: If it's undefined we shouldn't be mounting this component in the first place, if we want to handle undefined state (like loading) it should be reflected in the Props `data` type so TS can catch that

**Comment by krzysztofwolski** [comment]
Re console error: https://github.com/codex-team/editor.js/issues/1380 - soon there should be new release

File: src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (krzysztofwolski) [reply src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx:60]: Check is also done before mounting. 
Strange, when I removed any of those two lines (60 and 62) editor initializes every time I change tabs (description/attrs) so I would say that it works as should and those are needed. WDYT?

**Comment by krzysztofwolski** [comment]
Cypress seems to be failing because of test database - everything was green at 453c17b, after adding CHANGELOG tests started failing 🌵

**Review by jwm0 — APPROVED** [review]

**Review by orzechdev — APPROVED** [review]

File: src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply src/@next/components/atoms/RichTextEditorContent/RichTextEditorContent.tsx]: Right 🤔

**Pull request merged**